k8s/client.go

```diff
@@ -86,28 +71,37 @@ func New(kubeconfig string, resyncPeriod time.Duration) (Client, error) {
 		return nil, err
 	}
 
-	return &client{clientset: clientset, resyncPeriod: resyncPeriod}, nil
+	c := &client{
+		clientset:    clientset,
+		resyncPeriod: resyncPeriod,
+		store:        NewStore(clientset, resyncPeriod, stopCh),
+	}
+	return c, nil
 }
 
 func (c *client) GetAllIngresses() ([]*v1beta1.Ingress, error) {
 	return c.GetIngresses(nil)
 }
 
 func (c *client) GetIngresses(selector *NamespaceSelector) ([]*v1beta1.Ingress, error) {
-	if !c.namespaceController.HasSynced() {
```

Comment:
To avoid the ingress partial updates, we should include the check for the `ingressController` here as well.

Reply:
yeah - this is where it used to be, but was unintentionally removed in a previous PR...